sunpy/visualization/wcsaxes_compat.py

```diff
@@ -142,11 +154,23 @@ def wcsaxes_heliographic_overlay(axes):
     axes : `wcsaxes.WCSAxes` object.
         The `~wcsaxes.WCSAxes` object to create the HGS overlay on.
 
+    grid_spacing: float
+        Spacing for longitude and latitude grid.
+
     Returns
     -------
     overlay : wcsaxes overlay
         The overlay object.
     """
+
+    # Unpack spacing
+    if isinstance(grid_spacing, u.Quantity) and grid_spacing.size == 1:
+        lon_space = lat_space = grid_spacing
+    elif grid_spacing.size == 2:
+        lon_space, lat_space = grid_spacing
+    else:
+        raise ValueError("grid_spacing must be a Quantity or a tuple of two Quantites")
```

Comment:
so, or one quantity, or a tuple of two (which can be with different units, right?) That's probably why we cannot use the decorator...